src/coreComponents/physicsSolvers/contact/ContactSolverBase.cpp

```diff
@@ -0,0 +1,297 @@
+
+  this->getWrapper< string >( viewKeyStruct::discretizationString() ).
+    setInputFlag( InputFlags::FALSE );
+}
+
+
+void ContactSolverBase::postProcessInput()
+{
+  m_solidSolver = &this->getParent().getGroup< SolidMechanicsLagrangianFEM >( m_solidSolverName );
+  SolverBase::postProcessInput();
+}
+
+void ContactSolverBase::registerDataOnMesh( dataRepository::Group & meshBodies )
+{
+  meshBodies.forSubGroups< MeshBody >( [&] ( MeshBody & meshBody )
+  {
+    MeshLevel & meshLevel = meshBody.getMeshLevel( 0 );
+
+    ElementRegionManager & elemManager = meshLevel.getElemManager();
+    {
+      elemManager.forElementRegions< SurfaceElementRegion >( [&] ( SurfaceElementRegion & region )
+      {
+        region.forElementSubRegions< SurfaceElementSubRegion >( [&]( SurfaceElementSubRegion & subRegion )
+        {
+
+          subRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::dispJumpString() ).
+            setPlotLevel( PlotLevel::LEVEL_0 ).
+            reference().resizeDimension< 1 >( 3 );
+
+          subRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::deltaDispJumpString() ).
+            reference().resizeDimension< 1 >( 3 );
+
+          subRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::oldDispJumpString() ).
+            reference().resizeDimension< 1 >( 3 );
+
+          subRegion.registerWrapper< array2d< real64 > >( viewKeyStruct::tractionString() ).
+            setApplyDefaultValue( 0.0 ).
+            setPlotLevel( PlotLevel::LEVEL_0 ).
+            setRegisteringObjects( this->getName()).
+            setDescription( "An array that holds the tractions on the fracture." ).
+            reference().resizeDimension< 1 >( 3 );
+
+          subRegion.registerWrapper< array1d< integer > >( viewKeyStruct::fractureStateString() ).
+            setPlotLevel( PlotLevel::LEVEL_0 ).
+            setRegisteringObjects( this->getName()).
+            setDescription( "An array that holds the fracture state." );
+          initializeFractureState( meshLevel, viewKeyStruct::fractureStateString() );
+
+          subRegion.registerWrapper< array1d< integer > >( viewKeyStruct::oldFractureStateString() ).
+            setPlotLevel( PlotLevel::NOPLOT ).
+            setRegisteringObjects( this->getName()).
+            setDescription( "An array that holds the fracture state." );
+          initializeFractureState( meshLevel, viewKeyStruct::oldFractureStateString() );
+
+        } );
+      } );
+    }
+  } );
+}
+
+
+void ContactSolverBase::initializeFractureState( MeshLevel & mesh,
+                                                 string const & fieldName ) const
+{
+  GEOSX_MARK_FUNCTION;
+  ElementRegionManager & elemManager = mesh.getElemManager();
+
+  elemManager.forElementSubRegions< FaceElementSubRegion >( [&]( FaceElementSubRegion & subRegion )
+  {
+    if( subRegion.hasWrapper( viewKeyStruct::tractionString() ) )
```

Comment:
I don't think so. This must be an old check. It's not really for the solver I wrote though so I will check with @andrea-franceschini ?

Reply:
This wrapper was used to identify the `faceElementSubRegions` where I am solving Lagrangian contact mechanics, that is I want to avoid `faceElementSubRegions` of other solvers. Is there the possibility of having `faceElementSubRegions` with no `FractureState` defined on them?